### PR TITLE
Update-for-refactor

### PR DESCRIPTION
**Pull request by jmcook1186**
update information in docs for consistency with the refactored IF

### CODE REVIEW

**Review by narekhovhannisyan — APPROVED** [review]

File: docs/intro.md [hunk 1]
```diff
@@ -63,11 +63,11 @@ But now we are in the next evolution, to have software measurement be a mainstre
...
 
-We do provide a **standard library of models** built and maintained by the IF core team. These can be found in the [`if-models` Github repository](https://github.com/Green-Software-Foundation/if-models). You can install these into `if` by running `npm install https://github.com/Green-Software-Foundation/if-models` from the `if` project directory.
+We do provide a **standard library of plugins** built and maintained by the IF core team. These can be found in the [`if-models` Github repository](https://github.com/Green-Software-Foundation/if-models). You can install these into `if` by running `npm install https://github.com/Green-Software-Foundation/if-models` from the `if` project directory.
```
Comment (narekhovhannisyan) [inline docs/intro.md:68]: have we started renaming if-models to if-plugins? we should be consistent now with it

Comment (jmcook1186) [reply docs/intro.md:68]: yea i deliberately didn't update those URLs until we've confirmed the repo name changes are happening today

File: docs/developers/how-to-refine-plugins.md [hunk 2]
```diff
@@ -85,7 +84,7 @@ We use `zod` to validate data. Here's an example from our codebase:
 /**
  * Checks for required fields in input.
  */
-private validateInput(input: ModelParams) {
+private validateInput(input: pluginParams) {
```
Comment (manushak) [inline docs/developers/how-to-refine-plugins.md:87]: change function declaration to this one 
`const validateInput = (input: PluginParams) => {`

File: docs/developers/how-to-visualize-results.md [hunk 3]
```diff
@@ -20,24 +20,24 @@ If you need to visualize other output parameters you can either use **Grafana**
...
 description:
 tags:
 initialize:
-  models:
-    - name: simple-html-exporter
-      model: SimpleHtmlExporter
-      path: "@grnsft/if-models"
-graph:
+  plugins:
+    'simple-html-exporter'
```
Comment (manushak) [inline docs/developers/how-to-visualize-results.md:37]: at the end of the name add `:`

File: docs/major-concepts/manifest-file.md [hunk 4]
```diff
@@ -25,22 +25,23 @@ Anyone can download Impact Framework and execute a manifest file to verify the r
...
+Manifest files can be simple or very intricate, depending on the plugin pipeline you want to use and the complexity of your application. However, all manifest files conform to a basic structure that looks as follows:
 
 ```yaml
 name:
 description:
 tags:
 initialize:
-  models:
+  plugins:
     - name:
```
Comment (manushak) [inline docs/major-concepts/manifest-file.md:36]: remove `-` from the beginning

File: docs/major-concepts/manifest-file.md [hunk 5]
```diff
@@ -25,22 +25,23 @@ Anyone can download Impact Framework and execute a manifest file to verify the r
...
 description:
 tags:
 initialize:
-  models:
+  plugins:
     - name: 
-      model: 
+      method: 
       path: 
 graph:
```
Comment (manushak) [inline docs/major-concepts/manifest-file.md:39]: change to `tree`

File: docs/major-concepts/manifest-file.md [hunk 6]
```diff
@@ -55,16 +56,16 @@ The global metadata includes the `name`, `description` and `tags` that can be us
 The initialize section is where tyou define which plugins will be used in your manifest file and provide the global configuration for them. The required values are:
```
Comment (manushak) [inline docs/major-concepts/manifest-file.md:56]: typo in `tyou`

File: docs/major-concepts/manifest-file.md [hunk 7]
```diff
@@ -141,11 +145,11 @@ graph:
 
 ### Inputs
 
-Every component includes an `inputs` field that gets read into models as an array. `inputs` are divided into `observations`, each having a `timestamp` and a `duration`. Every `observation` refers to an element in `inputs` representing some snapshot in time.
+Every component includes an `inputs` field that gets read into plugins as an array. `inputs` are divided into `observations`, each having a `timestamp` and a `duration`. Every `observation` refers to an element in `inputs` representing some snapshot in time.
 
 Each plugin takes the `inputs` array and applies some calculation or transformation to each `observation` in the array.
 
-Observations can inlude any type of data, including human judgment, assumptions, other models, APIs, survey data or telemetry.
+Observations can inlude any type of data, including human judgment, assumptions, other plugins, APIs, survey data or telemetry.
```
Comment (manushak) [inline docs/major-concepts/manifest-file.md:152]: typo in `inlude`

File: docs/major-concepts/manifest-file.md [hunk 8]
```diff
@@ -171,16 +175,17 @@ name: e-mem
...
 tags: null
 initialize:
-  models:
+  plugins:
     - name: e-mem
-      path: "@grnsft/if-models"
-      model: EMemModel
+      path: "@grnsft/if-plugins"
+      method: EMem
 graph:
```
Comment (manushak) [inline docs/major-concepts/manifest-file.md:182]: rename to `tree`

File: docs/reference/cli.md [hunk 9]
```diff
@@ -1,29 +1,29 @@
...
 
-The `--ompl` flag is optional and is used for defining a path to save your output data. If you provide the `--ompl` command with a path, your output data will be saved as a `.yml` file to disk. If you omit this command, your output data will be displayed in the terminal.
+The `--output` flag is optional and is used for defining a path to save your output data. If you provide the `--output` command with a path, your output data will be saved as a `.yml` file to disk. If you omit this command, your output data will be displayed in the terminal.
 
-Here is an example of `--ompl` being used to define a path:
+Here is an example of `--output` being used to define a path:
 
 ```sh
-impact-engine --impl examples/impls/my-manifest.yml --ompl examples/ompls/my-outdata.yml
+impact-engine --manifest examples/my-manifest.yml --output examples/my-outdata.yml
```
Comment (manushak) [inline docs/reference/cli.md:28]: - `examples/my-manifest.yml` should be `examples/manifests/my-manifest.yml`
- `examples/my-outdata.yml` should be `examples/outputs/my-outdata.yml`

File: docs/reference/plugins.md [hunk 10]
```diff
@@ -1,34 +1,38 @@
...
+* [SCI](https://github.com/Green-Software-Foundation/if-plugins/blob/main/src/lib/sci/README.md): Calculates the software carbon intensity.
+* [SHELL](https://github.com/Green-Software-Foundation/if-plugins/blob/main/src/lib/shell/README.md) - A plugin that enables external code in any language to be run in a child process
+* [TDP-FINDER](https://github.com/Green-Software-Foundation/if-plugins/tree/main/src/lib/tdp-finder): Looks up the thermnal desig power for a given processor in a local database.
+* [Sum](https://github.com/Green-Software-Foundation/if-plugins/tree/main/src/lib/sum): a generic arithmetic plugin that allows you to sum any set of input parameters.
+* [Multiply](https://github.com/Green-Software-Foundation/if-plugins/tree/main/src/lib/multiply): a generic arithmetic plugin that allows you to multiply any set of input parameters.
+* [Coefficient](https://github.com/Green-Software-Foundation/if-plugins/tree/main/src/lib/coefficient): a generic arithmetic plugin that allows you to multiply any input value by a coefficient.
```
Comment (manushak) [inline docs/reference/plugins.md:28]: also, add links for
-  `e-net`
-  `mock-observations`
- `csv-export`

File: docs/reference/plugins.md [hunk 11]
```diff
@@ -1,34 +1,38 @@
...
-* [Boavizta](https://github.com/Green-Software-Foundation/if-unofficial-models/blob/main/src/lib/boavizta/README.md): Calculates energy and embodied carbon using the Boavizta APIs.
+
+### `if-unofficial-plugins`
+
+* [Azure importer](https://github.com/Green-Software-Foundation/if-unofficial-plugins/blob/main/src/lib/azure-importer/README.md): Grabs usage metrics from an Azure virtual machine, given user credentials and virtual machine details.
+* [Cloud Carbon Footprint](https://github.com/Green-Software-Foundation/if-unofficial-plugins/blob/main/src/lib/ccf/README.md): Calculates usage metrics using the Cloud Carbon Footprint APIs.
+* [WattTime](https://github.com/Green-Software-Foundation/if-unofficial-plugins/blob/main/src/lib/watt-time/README.md): WattTime is an external service for looking up grid emissions based on location.
+* [TEADS-CPU](https://github.com/Green-Software-Foundation/if-unofficial-plugins/blob/main/src/lib/teads-curve/README.md): Calculates the energy in kWh used by the CPU
```
Comment (manushak) [inline docs/reference/plugins.md:36]: `TEADS-CPU` rename to `TEADS-CURVE`
also, add `co2js`

File: docs/users/how-to-import-plugins.md [hunk 12]
```diff
@@ -2,58 +2,59 @@
...
+Plugins in these packages can then be invoked in an impl by providing their path in the plugin initialization, as shown in the following example:
 
 ```yaml
 name: if-demo
 description: demo pipeline
 tags:
 initialize:
-  models:
+  plugins:
     - name: azure-importer
```
Comment (manushak) [inline docs/users/how-to-import-plugins.md:24]: change 
- `- name: azure-importer` to `azure-importer:`
- `- name: cloud-instance-metadata` to `cloud-instance-metadata:`

File: docs/users/how-to-import-plugins.md [hunk 13]
```diff
@@ -2,58 +2,59 @@
...
-npm install https://github.com/Green-Software-Foundation/if-models
+npm install https://github.com/Green-Software-Foundation/if-plugins
 ```
 
 You'll need to provide the following fields:
 
-- `name`: the same name has to be used to refer to this model everywhere across the impl
-- `model`: the class name for your model, e.g. `AzureImporterModel` or `OutputModel`
-- `path`: the path to the model
+- `name`: the same name has to be used to refer to this plugin everywhere across the impl
```
Comment (manushak) [inline docs/users/how-to-import-plugins.md:41]: change `impl` to `manifest`

File: docs/users/how-to-import-plugins.md [hunk 14]
```diff
@@ -2,58 +2,59 @@
...
+name: plugin-demo
+description: loads plugin
 tags: null
 initialize:
-  models:
-    - name: my-model
-      model: OutputModel
-      path: https://github.com/my-repo/my-model
+  plugins:
+    - name: my-plugin
```
Comment (manushak) [inline docs/users/how-to-import-plugins.md:55]: should be `my-plugin:`

File: docs/users/how-to-write-impls.md [hunk 15]
```diff
@@ -4,27 +4,28 @@ sidebar_position: 4
...
+The basic structure of a manifest is as follows: 
 
 ```yaml
 name: 
 description: 
 tags: 
 initialize:
-  models:
+  plugins:
     - name:
```
Comment (manushak) [inline docs/users/how-to-write-impls.md:19]: remove `-`

File: docs/users/how-to-write-impls.md [hunk 16]
```diff
@@ -40,34 +41,35 @@ tags:
 
 ### Initialize
 
-The `initialize` fields are where you specify each individual model that will be initialized in your pipeline. The models can be initialized in any order, but can only be invoked elsewhere in the `impl` if they have been initialized first here. In each case, you will need to provide the name, path and model:
+The `initialize` fields are where you specify each individual plugin that will be initialized in your pipeline. The plugins can be initialized in any order, but can only be invoked elsewhere in the manifest if they have been initialized first here. In each case, you will need to provide the `name`, `path` and `method` (and `global-config` if your plugin requires it):
 
 ```yaml
 initialize:
-    - name: sci-m
+    sci-m
```
Comment (manushak) [inline docs/users/how-to-write-impls.md:48]: add `:` after sci-m

File: docs/users/how-to-write-impls.md [hunk 17]
```diff
@@ -90,176 +92,173 @@ inputs:
...
+- This means that `sci` will need to be preceded by `sci-m` and `sci-o` in the plugin pipeline.
 - In most cases, `sci-o` will have to be preceded by `sci-e` to ensure `energy` is available to be piped to `sci-o`.
-- The inputs to `sci-e` will most likely be coming from a model such as `teads-curve`.
-- The `sci` model also requires `functional-unit` information so it can convert the estimated `carbon` into a useful unit.
+- The inputs to `sci-e` will most likely be coming from a plugin such as `teads-curve` or `boavizta`.
+- The `sci` plugin also requires `functional-unit` information so it can convert the estimated `carbon` into a useful unit.
 - You may also wish to grab your `input` data by querying a metrics API on a virtual machine. 
 
-The example below gives you the full pipeline implemented in an `impl` (if you have an Azure virtual machine and put your credentials into a `.env` file, you can run this `impl` to retrieve your SCI score).
+The example below gives you the full pipeline implemented in an manifest.There are also several other executable example manifests in `if/examples/impls/functional/` that you can run for yourself.
```
Comment (manushak) [inline docs/users/how-to-write-impls.md:110]: `if/examples/impls/functional/` change to `if/examples/manifests/`

**Review by narekhovhannisyan — APPROVED** [review]

**Pull request merged**